### ISSUE TITLE
BoM Beta Website API

### ISSUE THREAD

**Issue by ArrEssJay** [bremor/bureau_of_meteorology#235]
Not so much an issue... I see the BoM beta website is using (yet another) more comprehensive JSON-based API. This appears to provide additional data e.g. air pressure where available, also at 10 minute intervals. I had a trawl through the issue list and saw no mention of this.

In all requests ensure user agent is set to something browser-like.

Selecting your location in https://beta.bom.gov.au/ will query a number of API endpoints either for some defaults or for your determined lat/lon position.

Get forecast/fire/flood/coastal district and nearest stations for a location:
`https://api.beta.bom.gov.au/apikey/v1/locations/places/search?coordinate={lon},{lat}&filter=type:place,nearby_type:bom_stn&radius=100000&website-sort=true&website-filter=true`

Get list of forecast districts/forecast IDs:
`https://beta.bom.gov.au/config/towns_places/towns-cities.json`

Daily Forecast:
`https://api.beta.bom.gov.au/apikey/v1/forecasts/daily-list/2024-07-28?place_id={FORECAST_ID}`

Detailed/Summary Forecast:
`https://api.beta.bom.gov.au/apikey/v1/forecasts/apikey/v1/forecasts/texts?aac={FORECAST_ID}&aac={FORECAST_ID_2}&timezone={timezone}`

Where timezone is in TZ format eg. 'Australia/Melbourne'

Single station latest observations:
`https://api.beta.bom.gov.au/apikey/v1/observations/latest/{bom_station_no}/atm/surf_air?include_qc_results=false`

List-of-Stations latest observations:
`https://api.beta.bom.gov.au/apikey/v1/observations/latest-list/atm/surf_air?bom_stn_num={bom_station_no_1}&bom_stn_num={bom_station_no_n}&include_qc_results=false`

Warnings:
`https://api.beta.bom.gov.au/apikey/v1/warnings/list?area_type=coordinate&area_code={lon},{lat}`

Sunrise/Sunset:
`https://api.beta.bom.gov.au/apikey/v1/forecasts/astro/{lon}/{lat}`

time-series map tiles for radar/temp etc are retrieved from ArcGIS Server:
`https://api.beta.bom.gov.au/apikey/v1/mapping/overlays/towns_and_cities/FeatureServer?f=pjson`
`https://api.beta.bom.gov.au/apikey/v1/mapping/basemaps/basemap_default/MapServer/tile/9/313/460`
`https://api.beta.bom.gov.au/apikey/v1/mapping/timeseries/wmts/1.0.0/atm_surf_air_precip_rate_1hr_total_mm_h/default/2024-07-28T14:00Z/GoogleMapsCompatible_BoM/8/25/29.png`

This appears to be behind a proxy - most ArcGIS/WMTS queries 404

There are likely more endpoints - I didn't perform an exhaustive exploration.

Will they change it? Will they block your IP if you use it? Who knows. I'll probably use Telegraf to poll the json and pass it into home assistant via MQTT for now, but perhaps it will prove useful in the custom component too.

**Comment by Makin-Things** [comment]
Thanks @ArrEssJay for the heads up.
This is very similar to the current api, but with lots more data made available. The current api is used by the mobile app, but for quite a long time there was also a beta version available at the same url that they are now reusing (they removed it about 2 years ago).
While they have the word beta in the api url I think it would not be wise to commit resources to developing new code to use it. It is a real shame that they are continuing down the path of polling for updates (if you watch in the dev tools you will see it polls the warning api every 60 seconds to get the same response 99.9999% of the time). Why not use websockets to push updates, it is 2024 after all so that is no longer bleeding edge tech.
I tried to use the give feedback option, but alas something as important as that does not function.

I will leave this issue open for others to share their thoughts.

**Comment by ArrEssJay** [comment]
I agree - I'd hold off until there is more evidence that its stable.
Clearly a significant amount of work is being done so it seems a reasonably
safe assumption that this is the direction they're headed.

The additional cloud level/visibility/wet bulb globe data seen at airports
might be of interest to some. For anyone unfamiliar, these are described in
the commercial 1 minute data product guide:
http://reg.bom.gov.au/catalogue/oneminuteobs.pdf

I wouldn't say the absence of a push mechanism is necessarily a bad thing,
so long as data is updated on a predictable schedule that polling can be
synchronised to. I'm not sure why the timing seems so lax. At least the
measurements are timestamped. 1 min data is available for a fee, over FTP.
Yeah.... 2024.

**Comment by tramperb** [comment]
There is also an API for the BoM's Space Weather data [https://sws-data.sws.bom.gov.au/](url)
It would be great to have access to, for example, aurora alerts in HA that are specific to Australia.

**Comment by nicois** [comment]
The beta site is now in production, so it is worth reviewing where things stand.

**Comment by systemtester** [comment]
Probably not until it becomes a legally available resource for public use (with documented services please BOM) would be my suggestion.  I had a look at this a while ago and if they introduce auth between front-end and back-end (they really should) into the current or new api then it all goes away so it would be a risky move for someone to rebuild this integration or create a new one against the richer/newer api without some certainty it wouldn't be removed.
